src/parser/GeoPoint.cpp

```diff
@@ -60,9 +60,10 @@ GeoPoint::T GeoPoint::toBitRepresentation() const {
 
 // _____________________________________________________________________________
 std::optional<GeoPoint> GeoPoint::parseFromLiteral(
-    const ad_utility::triple_component::Literal& value) {
-  if (value.hasDatatype() &&
-      value.getDatatype() == asNormalizedStringViewUnsafe(GEO_WKT_LITERAL)) {
+    const ad_utility::triple_component::Literal& value, bool checkDatatype) {
+  if ((value.hasDatatype() &&
+       value.getDatatype() == asNormalizedStringViewUnsafe(GEO_WKT_LITERAL)) ||
+      !checkDatatype) {
```

Comment:
If you want this as a micro-optimization, then you should put the `!checkDatatype` first, to make it short-circuit